src/hook/provider.tsx

```diff
@@ -1,25 +1,28 @@
 import React, { FC, useEffect, useRef, useState } from "react";
-import ToastContext from "./context";
+import ToastContext, { ToastType } from "./context";
 import Toast, { Props } from "../toast-container";
 
 type PropsWithChildren = Props & {
-    children: React.ReactNode;
+  children: React.ReactNode;
 }
 
+export let GlobalToast: ToastType
+
 const ToastProvider: FC<PropsWithChildren> = ({ children, ...props }) => {
-    const toastRef = useRef(null);
-    const [refState, setRefState] = useState({});
+  const toastRef = useRef(null);
+  const [refState, setRefState] = useState({});
 
-    useEffect(() => {
-        setRefState(toastRef.current as any);
-    }, []);
+  useEffect(() => {
+    setRefState(toastRef.current as any);
+    GlobalToast = toastRef.current as any
+  }, []);
 
-    return (
-        <ToastContext.Provider value={refState as any}>
-            {children}
-            <Toast ref={toastRef} {...props} />
-        </ToastContext.Provider>
-    );
+  return (
+    <ToastContext.Provider value={refState as any}>
```

Comment:
is there no better way to type this?

Reply:
Came from Library author, at this point state is set to {} which effectively silences compilation errors, there is but it would require bloating the code with more typesafety checking then necessary.